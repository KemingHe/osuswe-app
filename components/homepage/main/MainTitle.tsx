import type { NextFont } from 'next/dist/compiled/@next/font';
import localFont from 'next/font/local';
import type { JSX } from 'react';

const didotItalic: NextFont = localFont({
  // NOTE: Not a ts module, must use relative path.
  src: './fonts/Didot-Italic.ttf',
  display: 'swap',
});

export default function MainTitle(): JSX.Element {
  return (
    <div className="w-full flex flex-col bg-secondary p-4 gap-1">
      <h1
        className={`${didotItalic.className} flex flex-col justify-center items-start text-4xl gap-1`}
      >
        <span className="sr-only">Society of Women Engineers</span>
        <span aria-hidden="true">Society of</span>
        <span aria-hidden="true">Women Engineers</span>
      </h1>
      <p className="text-right">at The Ohio State University</p>
    </div>
  );
}
