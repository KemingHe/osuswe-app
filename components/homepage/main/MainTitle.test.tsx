import { type RenderResult, render, screen } from '@testing-library/react';

import MainTitle from '@/components/homepage/main/MainTitle';

vi.mock('next/font/local', () => ({
  __esModule: true,
  default: () => ({
    className: 'test-font-class',
  }),
}));

const hiddenTextSelector: string = 'span.sr-only';
const visibleTextSelector: string = 'span[aria-hidden="true"]';

describe('MainTitle component', () => {
  it('renders the hidden and visible text correctly', () => {
    render(<MainTitle />);

    // Verify that custom font class is present.
    const headingElement = screen.getByRole('heading');
    expect(headingElement).toHaveClass('test-font-class');

    // Verify that the hidden text is present.
    const hiddenText = screen.getByText('Society of Women Engineers', {
      selector: hiddenTextSelector,
    });
    expect(hiddenText).toBeVisible();

    // Verify that the visible text is present.
    const visibleTextPart1 = screen.getByText('Society of', {
      selector: visibleTextSelector,
    });
    const visibleTextPart2 = screen.getByText('Women Engineers', {
      selector: visibleTextSelector,
    });
    expect(visibleTextPart1).toBeVisible();
    expect(visibleTextPart2).toBeVisible();

    // Verify the static text.
    const staticText = screen.getByText('at The Ohio State University');
    expect(staticText).toBeVisible();
  });

  it('matches the snapshot', () => {
    const { asFragment }: RenderResult = render(<MainTitle />);
    expect(asFragment()).toMatchSnapshot();
  });
});
