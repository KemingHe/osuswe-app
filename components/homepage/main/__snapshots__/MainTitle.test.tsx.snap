// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainTitle component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full flex flex-col bg-secondary p-4 gap-1"
  >
    <h1
      class="test-font-class flex flex-col justify-center items-start text-4xl gap-1"
    >
      <span
        class="sr-only"
      >
        Society of Women Engineers
      </span>
      <span
        aria-hidden="true"
      >
        Society of
      </span>
      <span
        aria-hidden="true"
      >
        Women Engineers
      </span>
    </h1>
    <p
      class="text-right"
    >
      at The Ohio State University
    </p>
  </div>
</DocumentFragment>
`;
